Lib9c/Action/Sweep.cs

```diff
@@ -0,0 +1,195 @@
+                    .SetState(questListAddress, MarkChanged)
+                    .SetState(avatarAddress, MarkChanged)
+                    .SetState(context.Signer, MarkChanged);
+            }
+
+            var addressesHex = GetSignerAndOtherAddressesHex(context, avatarAddress);
+
+            if (!states.TryGetAvatarStateV2(context.Signer, avatarAddress, out var avatarState, out var migrationRequired))
+            {
+                throw new FailedLoadStateException($"{addressesHex}Aborted as the avatar state of the signer was failed to load.");
+            }
+
+            var sheets = states.GetSheets(
+                containQuestSheet: false,
+                containStageSimulatorSheets: false,
+                sheetTypes: new[]
+                {
+                    typeof(WorldSheet),
+                    typeof(StageSheet),
+                    typeof(MaterialItemSheet),
+                    typeof(StageWaveSheet),
+                    typeof(CharacterLevelSheet),
+                });
+
+            var worldSheet = sheets.GetSheet<WorldSheet>();
+            if (!worldSheet.TryGetValue(worldId, out var worldRow, false))
+            {
+                throw new SheetRowNotFoundException(addressesHex, nameof(WorldSheet), worldId);
+            }
+
+            if (stageId < worldRow.StageBegin ||
+                stageId > worldRow.StageEnd)
+            {
+                throw new SheetRowColumnException(
+                    $"{addressesHex}{worldId} world is not contains {worldRow.Id} stage: " +
+                    $"{worldRow.StageBegin}-{worldRow.StageEnd}");
+            }
+
+            if (!sheets.GetSheet<StageSheet>().TryGetValue(stageId, out var stageRow))
+            {
+                throw new SheetRowNotFoundException(addressesHex, nameof(StageSheet), stageId);
+            }
+
+            var worldInformation = avatarState.worldInformation;
+            if (!worldInformation.TryGetWorld(worldId, out var world))
+            {
+                worldInformation.AddAndUnlockNewWorld(worldRow, context.BlockIndex, worldSheet);
+            }
+
+            if (!world.IsStageCleared && stageId > world.StageClearedId)
+            {
+                throw new InvalidStageException(
+                    $"{addressesHex}Aborted as the stage ({worldId}/{stageId}) is not cleared; " +
+                    $"cleared stage: {world.StageClearedId}"
+                );
+            }
+
+            if (worldId == GameConfig.MimisbrunnrWorldId)
+            {
+                throw new InvalidWorldException(
+                    $"{addressesHex} [{worldId}] can't execute sweep action.");
+            }
+
+            var materialItemSheet = sheets.GetSheet<MaterialItemSheet>();
+
+            // check ap
+            if (apStoneCount > 0)
+            {
+                if (apStoneCount > UsableApStoneCount)
+                {
+                    throw new UsageLimitExceedException();
+                }
```

Comment:
Similar to `worldId` validation, validation of `apStoneCount` included in `plane value` is better to allocate `addressesHex` for log and then move it in order.